c_src/eleveldb.cc

```diff
@@ -353,14 +357,15 @@ ERL_NIF_TERM parse_open_option(ErlNifEnv* env, ERL_NIF_TERM item, leveldb::Optio
         }
         else if (option[0] == eleveldb::ATOM_COMPRESSION)
         {
-            if (option[1] == eleveldb::ATOM_TRUE)
-            {
+            if (option[1] == eleveldb::ATOM_ON || option[1] == eleveldb::ATOM_TRUE
+                || option[1] == eleveldb::ATOM_SNAPPY )
                 opts.compression = leveldb::kSnappyCompression;
-            }
```

Comment:
Personally, I do honestly - the whole "oops, that statement was supposed to be part of the "if/true" block but instead ended up just executing because I forgot a curly brace" bugs are nasty (see http://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/).